### PR TITLE
E2e tests for federated daemonset

### PR DESCRIPTION
**Pull request by mwielgus**
Based on federated secrets e2e tests.

cc: @quinton-hoole

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35067)

### CODE REVIEW

**Comment by mwielgus** [comment]
@quinton-hoole ping

**Comment by mwielgus** [comment]
@k8s-bot gci gke e2e test this

**Comment by quinton-hoole** [comment]
@k8s-bot Federation GCE e2e test this

File: test/e2e/federation-daemonset.go [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+)
+
+const (
+	FederatedDaemonSetName       = "federated-daemonset"
+	FederatedDaemonSetTimeout    = 60 * time.Second
+	FederatedDaemonSetMaxRetries = 3
+)
+
+// Create/delete daemonset api objects
+var _ = framework.KubeDescribe("Federation daemonsets [Feature:Federation12]", func() {
```
Comment (quinton-hoole) [inline test/e2e/federation-daemonset.go]: What's the "12" for here?

**Comment by quinton-hoole** [comment]
One nit, then LGTM.  Lets confirm that the tests pass on Jenkins before merging this.

**Comment by quinton-hoole** [comment]
@k8s-bot federation GCE e2e test this

**Comment by quinton-hoole** [comment]
@mwielgus Feel free to apply the LGTM label once the federation GCE e2e tests have passed on Jenkins.

**Comment by quinton-hoole** [comment]
@k8s-bot federation gce e2e test this

**Comment by quinton-hoole** [comment]
@mwielgus 
W1031 15:35:37.896] test/e2e/federation-daemonset.go:50: f.Client undefined (type *framework.Framework has no field or method Client)
W1031 15:35:37.897] test/e2e/federation-daemonset.go:56: f.Client undefined (type *framework.Framework has no field or method Client)
W1031 15:35:37.897] test/e2e/federation-daemonset.go:61: f.Client undefined (type *framework.Framework has no field or method Client)

**Comment by quinton-hoole** [comment]
Ping @mwielgus

**Comment by mwielgus** [comment]
@k8s-bot cvm gce e2e test this

**Pull request merged**